lint/registry.go

```diff
@@ -0,0 +1,88 @@
+package lint
+
+import (
+	"fmt"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// MakeRegistryFromAllFiles creates a *protoregistry.Files with all dependencies present in descs resolved.
+// Missing dependencies will be filled with placeholders. If there are cyclic dependencies, an error will
+// be returned.
+func MakeRegistryFromAllFiles(descs []*descriptorpb.FileDescriptorProto) (*protoregistry.Files, error) {
```

Comment:
Make this function private -- this will be replaced once upstream protoreflect support it.